### PR TITLE
mcompile: Add --verbose mode

### PR DESCRIPTION
**Pull request by ignatenkobrain**
Closes: https://github.com/mesonbuild/meson/issues/7352
Signed-off-by: Igor Raits <i.gnatenko.brain@gmail.com>

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
FWIW I don't know if VS does what I expect it to do since I have no place to test.

File: mesonbuild/mcompile.py [hunk 1]
```diff
@@ -74,8 +76,10 @@ def get_parsed_args_vs(options: 'argparse.Namespace', builddir: Path):
     
     if options.load_average:
         mlog.warning('Msbuild does not have a load-average switch, ignoring.')
+    if options.verbose:
+        cmd.append('/v:detailed')
```
Comment (xclaesse) [inline mesonbuild/mcompile.py:80]: I think we agreed that by default vs backend is already verbose enough, and thus do:
```
if not options.verbose:
  cmd.append('/v:minimal')
```

**Comment by TheQwertiest** [comment]
This should be implemented a bit different though. `detailed` is **too** detailed (as discussed in https://github.com/mesonbuild/meson/pull/7183). So, VS should have `minimal` by default and `normal` when `--verbose`. That will make ninja and msbuild to behave nearly identically.

**Comment by TheQwertiest** [comment]
On another note: `-v` switch is occupied by the main `meson` handler as a synonym for `--version`. So it should be removed from this handler.

File: mesonbuild/mcompile.py (hunk 1 above)
Comment (ignatenkobrain) [reply mesonbuild/mcompile.py]: Done.

File: mesonbuild/mcompile.py [hunk 2]
```diff
@@ -108,6 +112,11 @@ def add_arguments(parser: 'argparse.ArgumentParser') -> None:
         default='.',
         help='The directory containing build files to be built.'
     )
+    parser.add_argument(
+        '-v', '--verbose',
```
[the remark this answers is not in the document]
Comment (ignatenkobrain) [reply mesonbuild/mcompile.py]: done

**Comment by TheQwertiest** [comment]
The corresponding section of docs/Commands.md need to be updated with the output of 
```
COLUMNS=80 meson.py compile --help
```

**Comment by ignatenkobrain** [comment]
Thanks a lot for reviews - all suggestions should be applied by now.

**Comment by xclaesse** [comment]
That looks good to me, but I'll let someone actually using the vs backend confirm :)

**Comment by nirbheek** [comment]
Works great on the VS backend. But maybe we want a unit test? Here's sample output for the VS backend with the `common/7 mixed` test. Should be easy to write with it. We already have tests which do this sort of thing; see f.ex. `assertReconfiguredBuildIsNoop()`.

<details><summary>meson compile runs with and without --verbose</summary>

```
C:\projects\meson\meson.git\test cases\common\7 mixed>..\..\..\meson.py compile -C _build
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking whether solution needs to be regenerated.
  Everything is up-to-date, regeneration of build files is not needed.
  main.cc
  func.c
  prog@exe.vcxproj -> C:\projects\meson\meson.git\test cases\common\7 mixed\_build\.\prog.exe

C:\projects\meson\meson.git\test cases\common\7 mixed>..\..\..\meson.py compile -C _build --clean
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.


C:\projects\meson\meson.git\test cases\common\7 mixed>..\..\..\meson.py compile -C _build --verbose
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2020-06-18 23:19:48.
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" on node 1 (default targets).
     1>ValidateSolutionConfiguration:
         Building solution configuration "debug|x64".
       ValidateProjects:
         The project "RUN_TESTS" is not selected for building in solution configuration "debug|x64".
         The project "RUN_INSTALL" is not selected for building in solution configuration "debug|x64".
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (1) is building "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\REGEN.vcxproj" (2) on node 2 (default targets).
     2>InitializeBuildStatus:
         Creating "regen-temp\REGEN.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       CustomBuild:
         Checking whether solution needs to be regenerated.
         Everything is up-to-date, regeneration of build files is not needed.
       FinalizeBuildStatus:
         Deleting file "regen-temp\REGEN.tlog\unsuccessfulbuild".
         Touching "regen-temp\REGEN.tlog\REGEN.lastbuildstate".
     2>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\REGEN.vcxproj" (default targets).
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (1) is building "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\prog@exe.vcxproj" (3) on node 1 (default targets).
     3>InitializeBuildStatus:
         Creating "prog@exe\prog.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       ClCompile:
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\CL.exe /c /I"prog@exe" /I. /I..\ /ZI /JMC /nologo /W2 /WX- /diagnostics:column /Od /Ot /D _MBCS /Gm- /EHsc /RTC1 /MDd /GS /Gy /fp:precise /Zc:wchar_t /Zc:
         forScope /Zc:inline /Fo"prog@exe\main.cc.obj" /Fd"prog@exe\vc142.pdb" /Gd /TP /FC /errorReport:queue "/EHsc" ..\main.cc
         main.cc
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\CL.exe /c /I"prog@exe" /I. /I..\ /ZI /JMC /nologo /W2 /WX- /diagnostics:column /Od /Ot /D _MBCS /Gm- /EHsc /RTC1 /MDd /GS /Gy /fp:precise /Zc:wchar_t /Zc:
         forScope /Zc:inline /Fo"prog@exe\func.c.obj" /Fd"prog@exe\vc142.pdb" /Gd /TC /FC /errorReport:queue ..\func.c
         func.c
       Link:
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\link.exe /ERRORREPORT:QUEUE /OUT:".\prog.exe" /INCREMENTAL /NOLOGO kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ol
         e32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG:FULL
          /PDB:".\prog.pdb" /SUBSYSTEM:CONSOLE /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:".\prog.lib" /MACHINE:X64 "prog@exe\main.cc.obj"
         "prog@exe\func.c.obj"
         prog@exe.vcxproj -> C:\projects\meson\meson.git\test cases\common\7 mixed\_build\.\prog.exe
       FinalizeBuildStatus:
         Deleting file "prog@exe\prog.tlog\unsuccessfulbuild".
         Touching "prog@exe\prog.tlog\prog.lastbuildstate".
     3>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\prog@exe.vcxproj" (default targets).
     1>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (default targets).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.33

C:\projects\meson\meson.git\test cases\common\7 mixed>..\..\..\meson.py compile -C _build --clean --verbose
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2020-06-18 23:19:53.
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" on node 1 (Clean target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "debug|x64".
       ValidateProjects:
         The project "RUN_TESTS" is not selected for building in solution configuration "debug|x64".
         The project "RUN_INSTALL" is not selected for building in solution configuration "debug|x64".
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (1) is building "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\prog@exe.vcxproj" (2) on node 2 (Clean target(s)).
     2>_PrepareForClean:
         Deleting file "prog@exe\prog.tlog\prog.lastbuildstate".
     1>Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (1) is building "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\REGEN.vcxproj" (3) on node 1 (Clean target(s)).
     3>_PrepareForClean:
         Deleting file "regen-temp\REGEN.tlog\REGEN.lastbuildstate".
     3>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\REGEN.vcxproj" (Clean target(s)).
     2>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\prog@exe.vcxproj" (Clean target(s)).
     1>Done Building Project "C:\projects\meson\meson.git\test cases\common\7 mixed\_build\mixed C and C++.sln" (Clean target(s)).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.17

C:\projects\meson\meson.git\test cases\common\7 mixed>
```

</details>

**Comment by ignatenkobrain** [comment]
> But maybe we want a unit test?

Would be cool, but I won't have time for this. Sorry.

**Comment by ignatenkobrain** [comment]
@nirbheek does this look good?

**Comment by nirbheek** [comment]
lgtm. I just noticed you didn't add a `-v` option too. Why is that? :)

**Comment by TheQwertiest** [comment]
@nirbheek, meson core options use "-v" as a short hand for "--version"

**Pull request merged**